rmw_fastrtps_shared_cpp/src/rmw_take.cpp

```diff
@@ -75,14 +75,16 @@ _take(
   data.data = ros_message;
   data.impl = info->type_support_impl_;
   if (info->subscriber_->takeNextData(&data, &sinfo)) {
-    info->listener_->data_taken(info->subscriber_);
+    info->listener_->updateDataCount(info->subscriber_);
 
     if (eprosima::fastrtps::rtps::ALIVE == sinfo.sampleKind) {
       if (message_info) {
         _assign_message_info(identifier, message_info, &sinfo);
       }
       *taken = true;
     }
+  } else {
+    info->listener_->updateDataCount(info->subscriber_);
```

Comment:
this will be called in either way, maybe it can be outside of the if-else sentence. same below.

Reply:
I thought this before, this one could move after `if` so that use only one `data_taken`, but the latter one seems not because of a `return ret;` in the middle, and I don't want to update the original logic (such as `data_taken()` happened before `_assign_message_info`.